### PR TITLE
Ret 4104 - Welsh- Notifications emails for case progression

### PR DESCRIPTION
**Pull request by lvo700**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/RET-4104

### Change description ###

Welsh- Notifications emails for case progression

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [x] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/TseHelper.java [hunk 1]
```diff
@@ -221,19 +225,30 @@ public static String getReplyDocumentRequest(CaseData caseData, String accessKey
      * @return Personalisation For Response
      * @throws NotificationClientException Throw Exception
      */
-    public static Map<String, Object> getPersonalisationForResponse(CaseDetails caseDetails, byte[] document,
-                                                                    String citizenUrl)
+
+    public static Map<String, Object> getPersonalisationForResponse(CaseDetails caseDetails,
+                                                                    byte[] document,
+                                                                    String citizenUrl,
+                                                                    FeatureToggleService featureToggleService)
```
Comment (marcjudge) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/TseHelper.java:232]: Don't pass the service. Pass the value you want

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/PseRespondToTribunalService.java [hunk 2]
```diff
@@ -212,20 +217,30 @@ private String getHearingDate(CaseData caseData, SendNotificationType sendNotifi
...
-            emailService.sendEmail(notificationToClaimantTemplateId,
+            emailService.sendEmail(emailTemplate,
                 caseData.getClaimantType().getClaimantEmailAddress(),
                 buildPersonalisationNotify(caseDetails));
         }
     }
 
     private Map<String, String> buildPersonalisationNotify(CaseDetails caseDetails) {
         CaseData caseData = caseDetails.getCaseData();
+        boolean isWelsh = featureToggleService.isWelshEnabled()
```
Comment (marcjudge) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/PseRespondToTribunalService.java:234]: Duplicated code. You already established the value of isWelsh, use it

**Review by marcjudge — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/PseRespondToTribunalService.java (hunk 2 above)
Comment (lvo700) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/PseRespondToTribunalService.java]: Moved into its own method

**Review by jip1984 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/PseRespondToTribunalService.java [hunk 3]
```diff
@@ -212,23 +217,36 @@ private String getHearingDate(CaseData caseData, SendNotificationType sendNotifi
...
-            emailService.sendEmail(notificationToClaimantTemplateId,
+            emailService.sendEmail(emailTemplate,
                 caseData.getClaimantType().getClaimantEmailAddress(),
                 buildPersonalisationNotify(caseDetails));
         }
     }
 
     private Map<String, String> buildPersonalisationNotify(CaseDetails caseDetails) {
         CaseData caseData = caseDetails.getCaseData();
+        boolean isWelsh = isWelsh(caseData);
```
Comment (marcjudge) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/PseRespondToTribunalService.java:233]: This is still calling the featureToggleService twice in the same thread. On L220 you have the value of isWelsh. Pass this value to buildPersonalisationNotify on L227 then you don't need to get is value again here.

**Review by marcjudge — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/PseRespondToTribunalService.java (hunk 3 above)
Comment (lvo700) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/PseRespondToTribunalService.java]: Refactored

**Review by marcjudge — APPROVED** [review]

**Review by jip1984 — APPROVED** [review]

**Pull request merged**